src/components/guestsList/guestsList.js

```diff
@@ -0,0 +1,243 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { TableVariant } from '@patternfly/react-table';
+import { helpers } from '../../common';
+import { apiQueries, connect, reduxActions, reduxSelectors } from '../../redux';
+import { Loader } from '../loader/loader';
+import { inventoryListHelpers } from '../inventoryList/inventoryListHelpers';
+import { RHSM_API_QUERY_TYPES } from '../../types/rhsmApiTypes';
+import { Table } from '../table/table';
+
+/**
+ * A system inventory guests component.
+ *
+ * @augments React.Component
+ * @fires onUpdateGuestsData
+ * @fires onScroll
+ */
+class GuestsList extends React.Component {
+  state = { currentPage: 0, limit: 10, previousData: [] };
+
+  componentDidMount() {
+    this.onUpdateGuestsData();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { currentPage } = this.state;
+
+    if (currentPage !== prevState.currentPage) {
+      this.onUpdateGuestsData();
+    }
+  }
+
+  /**
+   * Call the RHSM APIs, apply filters.
+   *
+   * @event onUpdateGuestsData
+   */
+  onUpdateGuestsData = async () => {
+    const { currentPage, limit } = this.state;
+    const { getHostsInventoryGuests, query, id } = this.props;
+
+    if (id) {
+      const updatedQuery = {
+        ...query,
```

Comment:
Future:
Query isn't necessary based on the current API structure. We agree, we'll leave this in for now. 

Optional:
Place a "todo" to revisit removing "query" from "guestsList.js"